network/stream/v2/cursors_test.go

```diff
@@ -138,7 +138,19 @@ func TestNodesCorrectBinsDynamic(t *testing.T) {
 		}
 
 		// wait for the nodes to exchange StreamInfo messages
-		time.Sleep(100 * time.Millisecond)
+		wantCursorsCount := 17
+		for i := 499; i >= 0; i-- { // wait time 5s
+			time.Sleep(10 * time.Millisecond)
+			count1 := nodeRegistry(sim, nodeIDs[0]).getPeer(nodeIDs[1]).cursorsCount()
+			count2 := nodeRegistry(sim, nodeIDs[1]).getPeer(nodeIDs[0]).cursorsCount()
+			if count1 >= wantCursorsCount && count2 >= wantCursorsCount {
```

Comment:
Should the count be equal between the nodes?

Reply:
Yes, they should. This is just the wait loop in order to validate the exact bins later in this test.